src/engraving/dom/tie.cpp

```diff
@@ -501,7 +501,11 @@ bool Tie::isCrossStaff() const
 {
     const Note* startN = startNote();
     const Note* endN = endNote();
+    const Chord* startChord = startN ? startN->chord() : nullptr;
+    const Chord* endChord = endN ? endN->chord() : nullptr;
+    const staff_idx_t staff = staffIdx();
 
-    return (startN && startN->chord()->staffMove() != 0) || (endN && endN->chord()->staffMove() != 0);
+    return (startChord && (startChord->staffMove() != 0 || startChord->vStaffIdx() != staff))
+           || (endChord && (endChord->staffMove() != 0 || endChord->vStaffIdx() != staff));
```

Comment:
Could you explain why we need to add the `chord->vstaffIdx() != staff` condition? At a first glance it looks redundant to me, cause it can only be true if at least one chord has `staffMove() != 0`. Or is there a case that I'm not thinking of?